src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/ConstantsSummaryWriterImpl.java

```diff
@@ -97,20 +96,16 @@ public Content getContentsHeader() {
     }
 
     @Override
-    public void addLinkToPackageContent(PackageElement pkg,
-            Set<PackageElement> printedPackageHeaders, Content content) {
+    public void addLinkToPackageContent(String abbrevPackageName, Content content) {
         //add link to summary
         Content link;
-        if (pkg.isUnnamed()) {
+        if (abbrevPackageName.isEmpty()) {
             link = links.createLink(HtmlIds.UNNAMED_PACKAGE_ANCHOR,
                     contents.defaultPackageLabel, "");
         } else {
-            String parsedPackageName = utils.parsePackageName(pkg);
-            Content packageNameContent = Text.of(parsedPackageName + ".*");
-            link = links.createLink(DocLink.fragment(parsedPackageName),
+            Content packageNameContent = Text.of(abbrevPackageName + ".*");
```

Comment:
Separately. This is somewhat surprising to see that javadoc has been using `.*` to denote "subpackages" (or "related packages" as we call them elsewhere), whereas Java import statements use `.*` to denote immediately enclosed classes and interfaces, or static members thereof.